YACS-RCOS/yacs.n
# Automatically Switch to Dark Mode

**Issue**
> closes #408

**Database Changes/Migrations**
Untouched.

**Test Modifications**
Untouched.

**UI Modifications**
1. Removed the `switch` property of `b-form-checkbox`.
2. Moved out the moon icon from `#darkmode-toggle-form`.
3. Deleted the `<div>` near the moon icon.
4. Added `@dblclick` event to the moon icon.
5. Added two toast notifications.

**Test Procedure**
1. Double-click the moon icon to remove `darkMode` cookie (follow device's theme):
  The checkbox should in `indeterminate` state. 
2. Change the system theme to dark/light, the page will change color scheme *immediately*.
3. Click on the checkbox to specify dark/light mode.
  The checkbox should in `checked` state if the page is in dark mode, and `unchecked` for light mode.
  The color scheme should follow the checkbox even if the system color is changed.
4. Double-click the moon icon to follow the device's theme again, and remove cookie.

**Photos**
![Indeterminate](https://user-images.githubusercontent.com/11363469/110339030-a8ad1d80-8062-11eb-84f4-b518a26a9473.png)
![Checked](https://user-images.githubusercontent.com/11363469/110339097-bc588400-8062-11eb-8897-9edd45f588e1.png)
![Unchecked](https://user-images.githubusercontent.com/11363469/110339165-ce3a2700-8062-11eb-965d-1c0ac08d889f.png)

**Additional Info**
Modified, added some functions in `store/index.js`, which may impact the current usage of cookies.
If it has significant impact, the previous commit `d50be64` is a brutal force implementation, which does not change any current functions.

**Current Status**
~Working on automatically change color scheme without refreshing.~
~This works best with pure CSS, which needs refactoring.~
Never mind, finished...
